pkg/dns/ibm/public/cis_provider.go

```diff
@@ -161,7 +161,6 @@ func (p *Provider) createOrUpdateDNSRecord(record *iov1.DNSRecord, zone configv1
 	dnsName := strings.TrimSuffix(record.Spec.DNSName, ".")
 	listOpt.SetName(dnsName)
 	for _, target := range record.Spec.Targets {
-		listOpt.SetContent(target)
 		result, response, err := dnsService.ListAllDnsRecords(listOpt)
```

Comment:
Did you consider, if the target never changes, do we still need to loop on targets?  We definitely only need to make the call to `ListAllDnsRecords` once, outside the loop.